app/src/main/java/jet/Jet.java

```diff
@@ -25,6 +20,31 @@
  */
 public class Jet{
 
+    public static class Builder{
+        //Required parameters
+        private float mRadius;
+        private Position mSelfPos;
+        private JetAnimation mAnimation;
+        private CircleCollider collider;
+
+        //optional parameters
+        protected float mHealth = 100;
+        protected float mMaxSpeed = 20;
+        protected boolean mHasDestination = false;
+        protected int mGunType = Gun.GUN_TYPE_DEFAULT;
+        protected int mBulletStyle = Bullet.BULLET_STYLE_DEFAULT;
+
+
+        public Builder(float xPosition, float yPosition,float radius, int animationType){
```

Comment:
Builder shouldn't enforce parameters using constructor. Instead, for every missing non-optional parameter, it should throw an exception for the specific field. @Jelly-Yu